docs/developerguide.rst

```diff
@@ -9,4 +9,44 @@ understand REANA source code and contribute to it.
 Debugging
 ---------
 
-We use ``wdb``.
+Assuming that REANA is being run in development mode following
+`reana-cluster documentation
+<http://reana-cluster.readthedocs.io/en/latest/developerguide.html#deploying-latest-master-branch-versions>`_,
+to debug a component we need to rebuild its Docker image in debug mode. Let say
+we are going to debug `reana-server`:
+
+.. code-block:: console
+
+   $ # if we are not connected to the minikube docker daemon we should do it
+   $ eval "$(minikube docker-env)"
+   $ docker build . --build-arg DEBUG=true -t reanahub/reana-server:v0.1.0
+
+Set the breakpoint in your current project code using your preferred editor and
+add `import wdb; wdb.set_trace()` instruction, for instance in the
+`GET /api/analyses` endpoint. Then, we kill `reana-server` pod to update the
+source code and we query the mentioned endpoint:
+
+.. code-block:: console
+
+   $ kubectl delete pod --selector=app="server"
+   $ curl "192.168.99.100:30659/api/analyses?organization=default&user=00000000-0000-0000-0000-000000000000"
+
+After doing that we can go to the `wdb` dashboard (you can get ``wdb`` address using `reana-cluster get wdb <http://reana-cluster.readthedocs.io/en/latest/cliapi.html#reana-cluster-get>`_).
+
+.. image:: /_static/wdb-active-sessions.png
+
+And finally select the debugging session.
+
+.. image:: /_static/wdb-debugging-ui.png
+
+
+Limitations
+~~~~~~~~~~~
+
+Live code auto reload (such as Flask auto reload when ``FLASK_DEBUG`` is set)
+is not currently supported due to ``minikube mount`` limitations (it
+continuously updates files ``modify`` field so auto reload is triggered
```

Comment:
@hjhsalo could you please confirm this? if it also happens to you then we need to merge https://github.com/reanahub/reana-cluster/pull/48

You can quickly check it by having a [development cluster](http://reana-cluster.readthedocs.io/en/latest/developerguide.html#deploying-latest-master-branch-versions) and running:
```console
$ minikube ssh
> watch 'stat /code/reana-server/setup.py'
  File: '/code/reana-server/setup.py'
  Size: 3286            Blocks: 7          IO Block: 8192   regular file
Device: 25h/37d Inode: 8591305187  Links: 1
Access: (0644/-rw-r--r--)  Uid: (4294967294/ UNKNOWN)   Gid: (4294967294/ UNKNOWN)
Access: 1970-01-01 00:00:00.425043499 +0000
Modify: 2018-01-17 10:08:46.425043499 +0000
Change: 2018-01-17 10:08:46.425043499 +0000
 Birth: -
```

You should see that `Access`, `Change` and `Modify` change. Also if you have an idea on how to prevent this it would be helpful since the impact of this issue means adding an extra step to the debugging process. thanks 🙃

Reply:
I couldn't reproduce this behavior of `Access`, `Change` and `Modify` values changing. Initially I thought that the same thing would happen.

Below are the steps I did:
- I build `reana-server` with DEBUG-flag with Docker Client attached to Docker daemon of Minikube VM.
- I cloned all REANA components to `sources`-folder and mounted that folder to Minikube VM on path `/code`.
- I ran `reana-cluster -f ./reana_cluster/configurations/reana-cluster-dev.yaml init` and checked that all the components were deployed and their status is `Running` with `kubectl get pods`
- I double checked that Pod / Container where `reana-server`is running has environment variable `FLASK_DEBUG=1` set (`kubectl exec -it <name-reana-server-pod> bash` + `env | grep DEBUG`).

If I follow logs of `reana-server` Pod / Container I can see the `Restarting with stat` message and I can trigger a restart by e.g. `touch ./reana-server/rest/ping.py`.

I would say that `minikube ssh` isn't touching the files at least on my system (Ubuntu 16.04.XX LTS). Version of `minikube` binary is v0.25.0


I suggest that we check IRL when you are back at the office.

Reply:
I have upgraded to `minikube` `0.25.0` and indeed the problem with the files continuously getting updated disappeared 👍. Still need to test if the rest works.